src/swiftlet/azext_swiftlet/generated/_params.py

```diff
@@ -0,0 +1,82 @@
+# Changes may cause incorrect behavior and will be lost if the code is
+# regenerated.
+# --------------------------------------------------------------------------
+# pylint: disable=too-many-lines
+# pylint: disable=too-many-statements
+
+from azure.cli.core.commands.parameters import (
+    tags_type,
+    resource_group_name_type,
+    get_location_type
+)
+from azure.cli.core.commands.validators import get_default_location_from_resource_group
+from azext_swiftlet.action import (
+    AddSwiftletVirtualMachineCreatePorts,
+    AddSwiftletVirtualMachineUpdatePorts
+)
+
+
+def load_arguments(self, _):
+
+    with self.argument_context('swiftlet virtual-machine list') as c:
+        c.argument('resource_group_name', resource_group_name_type)
+
+    with self.argument_context('swiftlet virtual-machine show') as c:
+        c.argument('resource_group_name', resource_group_name_type)
+        c.argument('vm_name', type=str, help='The name of the virtual machine.', id_part='name')
+
+    with self.argument_context('swiftlet virtual-machine create') as c:
+        c.argument('resource_group_name', resource_group_name_type)
+        c.argument('vm_name', type=str, help='The name of the virtual machine.')
+        c.argument('tags', tags_type)
+        c.argument('location', arg_type=get_location_type(self.cli_ctx),
+                   validator=get_default_location_from_resource_group)
+        c.argument('swiftlet_bundle_sku', type=str, help='Specifies the Swiftlet bundle of this virtual machine (which '
+                   'specifies the selected tier of memory, processing, and storage).')
+        c.argument('swiftlet_image_id', type=str, help='The image ID to use. The image "platform" must match the '
+                   '"supportedImagePlatform" of the specified swiftletBundleSku.')
+        c.argument('username', type=str, help='The username for connecting the the virtual machine.')
+        c.argument('ssh_public_key', type=str, help='The SSH public key used to connect to this virtual machine. Only '
+                   'supported on Linux images. If specified on a Windows image, an error will be returned.')
+        c.argument('password', type=str, help='The password for connecting to this Swiftlet. If the image platform '
+                   'type is "linux", this is optional if sshPublicKey is set. If the image platform type is "windows", '
+                   'this is required.')
+        c.argument('ports', action=AddSwiftletVirtualMachineCreatePorts, nargs='*', help='The ports on which inbound '
+                   'traffic will be allowed.')
+        c.argument('startup_script', type=str, help='An inline script that will run upon startup of the virtual '
+                   'machine.')
+
+    with self.argument_context('swiftlet virtual-machine update') as c:
+        c.argument('resource_group_name', resource_group_name_type)
+        c.argument('vm_name', type=str, help='The name of the virtual machine.', id_part='name')
```

Comment:
does it support id? name should be enough right, since subcommand is vm.

Reply:
```
Resource Id Arguments
    --ids               : One or more resource IDs (space-delimited). It should be a complete
                          resource ID containing all information of 'Resource Id' arguments. You
                          should provide either --ids or other 'Resource Id' arguments.
    --name -n           : The name of the virtual machine.
    --resource-group -g : Name of resource group. You can configure the default group using `az
                          configure --defaults group=<name>`.
    --subscription      : Name or ID of subscription. You can configure the default subscription
                          using `az account set -s NAME_OR_ID`.
```